### ISSUE TITLE
Fix nn_system -> adapt_returnn_config_for_recog

### ISSUE THREAD

**Issue by christophmluscher** [rwth-i6/i6_experiments#40]
`nn_system->adapt_returnn_config_for_recog`

I think if this function is needed at all there are severe flaws in how we are building our configs. I am trying to solve the same issues right now for the CTC setup. I will give you am update when I have a clean solution for this.

_Originally posted by @JackTemaki in https://github.com/rwth-i6/i6_experiments/pull/30#discussion_r771418963_

**Comment by christophmluscher** [comment]
- [ ] add recursion _(If layer['class'] == 'subnetwork or for rec layers with nontrivial units, you would have to recursively descent into the unit and adapt also the inner layers.)_
- [ ] target does not need to be dropped
- [ ] `lstmp` as user error
- [ ] "cropped" in network: remove hardcoded part and replace with arbitrary user defined functions

**Comment by albertz** [comment]
This is obsolete with `returnn_common.nn.hybrid_hmm.IHybridHMM`, right?

**Comment by JackTemaki** [comment]
This is obsolete even without `returnn_common`, it is just a matter of adapting old setups. But the function is definitely not needed in a new setup no matter if you use `returnn_common` or not.

**Comment by albertz** [comment]
Is any work then needed? So this issue can be closed?

I was specifically referring to the concept of adapting a config (net dict) for recog. Such concept, such adapting logic is not needed when we have a clean interface which generates the config (net dict) directly either for train and recog. Then there is no need for any adaptation.

The current `IHybridHMM` interface looks like:
```python
class IHybridHMM(nn.Module):
  """
  Hybrid NN-HMM interface
  """

  encoder: EncoderType
  out_dim: nn.Dim

  def __call__(self, source: nn.Tensor, *,
               state: Optional[nn.LayerState] = None,
               train: bool = False, targets: Optional[nn.Tensor] = None) -> Tuple[nn.Tensor, Optional[nn.LayerState]]:
    """
    :param source: [B,T,in_dim], although not necessarily in that order. we require that time_dim_axis is set.
    :param state: previous state
    :param train: if set, it assumes we have given a target, and it will add losses
    :param targets: only used for training. [B,T] -> out_dim sparse targets.
    :return: (posteriors, final state),
      posteriors as log-prob, as [B,T',out_dim], although dims not necessarily in that order.
    """
    raise NotImplementedError
```

**Comment by JackTemaki** [comment]
I would keep it open until this is removed from the code, but yes no adaptation is needed.

**Comment by JackTemaki** [comment]
Solved with #95
